Classes/Routing/FrontendNodeRoutePartHandler.php

```diff
@@ -97,4 +98,59 @@ protected function resolveRoutePathForNode(NodeInterface $node)
 
         return trim($dimensionsUriSegment . $requestPath, '/') . $nodeContextPathSuffix;
     }
+    
+    /**
+     * Returns the initialized node that is referenced by $requestPath, based on the node's
+     * "uriPathSegment" property.
+     *
+     * Note that $requestPath will be modified (passed by reference) by buildContextFromRequestPath().
+     *
+     * @param string $requestPath The request path, for example /the/node/path@some-workspace
+     * @return NodeInterface
+     * @throws \Neos\Neos\Routing\Exception\NoWorkspaceException
+     * @throws \Neos\Neos\Routing\Exception\NoSiteException
+     * @throws \Neos\Neos\Routing\Exception\NoSuchNodeException
+     * @throws \Neos\Neos\Routing\Exception\NoSiteNodeException
+     * @throws \Neos\Neos\Routing\Exception\InvalidRequestPathException
+     */
+    protected function convertRequestPathToNode($requestPath)
+    {
+
+        // Custom check if request path is a backend request and skip further matching.
+        if ($requestPath === 'neos' || strpos($requestPath, 'neos/') === 0) {
+            throw new Exception\NoSuchNodeException(sprintf('No match possible because "%s" is a backend request path', $requestPath), 1346949857);
+        }
+
+        $contentContext = $this->buildContextFromRequestPath($requestPath);
+        $requestPathWithoutContext = $this->removeContextFromPath($requestPath);
+
+        $workspace = $contentContext->getWorkspace();
+        if ($workspace === null) {
+            throw new Exception\NoWorkspaceException(sprintf('No workspace found for request path "%s"', $requestPath), 1346949318);
+        }
+
+        $site = $contentContext->getCurrentSite();
+        if ($site === null) {
+            throw new Exception\NoSiteException(sprintf('No site found for request path "%s"', $requestPath), 1346949693);
+        }
+
+        $siteNode = $contentContext->getCurrentSiteNode();
+        if ($siteNode === null) {
+            $currentDomain = $contentContext->getCurrentDomain() ? 'Domain with hostname "' . $contentContext->getCurrentDomain()->getHostname() . '" matched.' : 'No specific domain matched.';
+            throw new Exception\NoSiteNodeException(sprintf('No site node found for request path "%s". %s', $requestPath, $currentDomain), 1346949728);
+        }
+
+        if ($requestPathWithoutContext === '') {
+            $node = $siteNode;
+        } else {
+            $relativeNodePath = $this->getRelativeNodePathByUriPathSegmentProperties($siteNode, $requestPathWithoutContext);
+            $node = ($relativeNodePath !== false) ? $siteNode->getNode($relativeNodePath) : null;
+        }
+
+        if (!$node instanceof NodeInterface) {
+            throw new Exception\NoSuchNodeException(sprintf('No node found on request path "%s"', $requestPath), 1346949857);
```

Comment:
The Exception code was used in line 121 already ;)

Reply:
yes, I thought using the same exception is a good idea because the exception purpose  (NoSuchNodeException) is the same and introducing a new one is just overhead. I can change that - any suggestion for naming? 

Explanation: Bypassing the mapping process results a big performance improvement.

Reply:
@davidspiola Using the same exception is totally fine. but the exception *code* should be unique to make this case distinguishable. `1346949857` was used twice ;)

Reply:
ok, I’ll change it